torchvision/datasets/triplet.py

```diff
@@ -0,0 +1,86 @@
+import torch
+import torch.utils.data as data
+
+import math
+
+
+def get_class_samples(num_classes, samples):
+    """Bins samples with respect to classes
+
+    Args:
+        num_classes (int): number of classes
+        samples (list[tuple]): (sample, class_idx) tuples
+
+    Returns:
+        list[list]: Bins of sample paths, binned by class_idx
+    """
+    class_samples = [[] for _ in range(num_classes)]
+    for sample_path, class_idx in samples:
+        class_samples[class_idx].append(sample_path)
+
+    return class_samples
+
+
+def generate_triplet(class_samples):
+    """Generates a triplet from bins of samples
+
+    Args:
+        class_samples (list[list]): bins of samples, binned by class
+        num_triplets (int): number of triplets to be generated
+
+    Returns:
+        list[tuple]: triplets of the form (anchor, positive, negative)
+    """
+    pos_cls, neg_cls = torch.multinomial(torch.ones(len(class_samples)), 2).tolist()
```

Comment:
I don't think we need to convert this to a list, as it will bring additional overhead, just to be wrapped again into a tensor in the next call.

What about using `torch.unbind` to split the single tensor into two?

Reply:
`torch.unbind` doesn't seem work well with `defaultdict` since `defaultdict` doesn't cast the tensor into an `int` for indexing. Will `torch.unbind` followed by `.item()` during indexing be better, or will it be the same as `.tolist()`?

Reply:
I misunderstood what was the output of this line (I thought `pos_cls` was a Tensor or `len(class_samples)`).

This is fine